pilot/pkg/serviceregistry/kube/controller/ambient/statusqueue/queue.go

```diff
@@ -41,10 +51,20 @@ type statusItem struct {
 }
 
 // NewQueue builds a new status queue.
-func NewQueue() *StatusQueue {
+func NewQueue(active *model.ActiveNotifier) *StatusQueue {
```

Comment:
consider renaming active to `notifier` instead. It would make adding the handler a little more readable to my mind.